doc/build-helpers/trivial-build-helpers.chapter.md

```diff
@@ -557,14 +557,18 @@ This creates a derivation with a directory structure like the following:
 
 ## `writeReferencesToFile` {#trivial-builder-writeReferencesToFile}
 
-Writes the closure of transitive dependencies to a file.
+Deprecated. Use [`writeClosure`](#trivial-builder-writeClosure) instead.
 
-This produces the equivalent of `nix-store -q --requisites`.
+## `writeClosure` {#trivial-builder-writeClosure}
+
+Given a list of [store paths](https://nixos.org/manual/nix/stable/glossary#gloss-store-path) (or string-like expressions coercible to store paths), it writes the combined [closure](https://nixos.org/manual/nix/stable/glossary#gloss-closure) of them to a text file.
```

Comment:
```suggestion
Builds a line-separated file with all of the [store paths](https://nixos.org/manual/nix/stable/glossary#gloss-store-path) in the [closure](https://nixos.org/manual/nix/stable/glossary#gloss-closure) of a list of root paths
```

Reply:
Could also make it


```suggestion
Builds a line-separated file with [references](https://nixos.org/manual/nix/stable/glossary#gloss-reference) to all of the [store paths](https://nixos.org/manual/nix/stable/glossary#gloss-store-path) in the [closure](https://nixos.org/manual/nix/stable/glossary#gloss-closure) of a list of root paths
```


Idk

Reply:
"root paths" is kind of informal though

Reply:
@fricklerhandwerk sorry to ping but we're discussing names and documentation, would you like to share an opinion?

Reply:
Thank you for helping!

"References of all store paths in the closure" is equivalent to "the closure". I think it might be confusing to mention *reference* here.

Reply:
```suggestion
Given a list of [store paths](https://nixos.org/manual/nix/stable/glossary#gloss-store-path) (or string-like expressions coercible to store paths), write their  collective [closure](https://nixos.org/manual/nix/stable/glossary#gloss-closure) to a text file.
```

I (strongly) recommend using imperative in reference documentation, primarily for brevity. "Do this, do that."

@SomeoneSerge the original structure was actually better because the prose matches the functional interface: given <input> make <output>.

@ShamrockLee [to combine](https://en.wiktionary.org/wiki/combine#Etymology) means "to put together two things". Since the input is a list, the word should be along the lines of [to compose](https://en.wiktionary.org/wiki/compose#Etymology) or [to collect](https://en.wiktionary.org/wiki/collect#Etymology_1).

Reply:
Yes, "given x, do y" is actually my preference too; what I intended with the suggestion is to specify the format ("line-separated") and get rid of the "combined closure".

On the latter note, I'd just drop "combined": "the closure of a set of paths" is already meaningful, there's no need for extra adjectives. The extra adjectives can also be harmful if they are perceived as technical terms which they are not